subsys/bluetooth/controller/Kconfig

```diff
@@ -86,13 +123,41 @@ config SDC_RX_STACK_SIZE
 	  Size of the receiving thread stack, used to retrieve HCI events and
 	  data from the controller.
 
-# The SoftDevice Controller library variants are defined in nrfxlib, here we redefine
-# the choice to 'import' them, so they appear in the same menu as the rest.
-
 choice SOFTDEVICE_CONTROLLER_VARIANT
 	prompt "SoftDevice Controller variant"
-	bool
+	default SOFTDEVICE_CONTROLLER_MULTIROLE if ((BT_OBSERVER && BT_BROADCASTER) || \
+						     BT_CTLR_LLPM || \
+						     BT_CTLR_ADV_EXT || \
+						     BT_CTLR_PHY_CODED || \
+						     BT_CTLR_ADV_PERIODIC || \
+						     SOC_NRF5340_CPUNET)
+	default SOFTDEVICE_CONTROLLER_CENTRAL if BT_OBSERVER
+	default SOFTDEVICE_CONTROLLER_PERIPHERAL if BT_BROADCASTER
+	help
+	  Select a SoftDevice Controller variant.
 
-endchoice
+config SOFTDEVICE_CONTROLLER_PERIPHERAL
```

Comment:
Why? We already have `BT_PERIPHERAL` and `BT_CENTRAL`?

Reply:
@cvinayak @alwa-nordic @joerchan any thoughts here?

Reply:
@carlescufi . These are the names of the libraries. See the help text

Reply:
I guess they're badly named, but they are three distinct libraries, with the central/peripheral being stripped-out size-optimized versions and the multirole being a full-featured version

Reply:
I see, fair enough.

Can we please maintain some consistency here? We chose `CTLR` a long time ago for the upstream one, can we try to do the same here? Maybe:

`BT_CTLR_SD_<>`

I mean you are not even prefixing with `BT_` here

Reply:
I mean we could either rename or move the logic to cmake, but then we lose the ability to force one or the other. Maybe having a check on BT_CENTRAL and BT_PERIPHERAL, and exposing two choices: SIZE_OPTIMIZED and FULL_FEATURED would be better ?

Reply:
I don't mind either way, as long as we maintain consistency. Please add @tejlmand as a reviewer in the follow-up PR.

Reply:
Created https://github.com/nrfconnect/sdk-nrf/pull/6107